docling_core/transforms/chunker/base.py

```diff
@@ -7,24 +7,40 @@
 from abc import ABC, abstractmethod
 from typing import Iterator, Optional
 
-from pydantic import BaseModel
+from pydantic import BaseModel, model_validator
 
 from docling_core.types import BoundingBox, Document
 
 
+def _create_path(pos: int, path_prefix: str = "main-text") -> str:
+    return f"#/{path_prefix}/{pos}"
+
+
 class Chunk(BaseModel):
     """Data model for Chunk."""
 
     path: str
     text: str
+    heading: Optional[str] = None
+
+    @model_validator(mode="before")
```

Comment:
I would use a field validator, since you are validating a specific field of a model and you don't need the other data.
In addition, the mode should be `after`, since you can leverage the pydantic control on the type (`str`) first and then continue with your string assumption in `if path.startswith(`

Reply:
See https://github.com/DS4SD/docling-core/pull/39.